src/ape/managers/project/types.py

```diff
@@ -1,20 +1,96 @@
-import json
+from contextlib import contextmanager
 from pathlib import Path
 from typing import Any, Dict, List, Optional
 
 import yaml
-from ethpm_types import PackageManifest
+from ethpm_types import ContractType, PackageManifest, Source
 from ethpm_types.utils import compute_checksum
 
 from ape.api import ProjectAPI
 from ape.logging import logger
 from ape.managers.config import CONFIG_FILE_NAME as APE_CONFIG_FILE_NAME
-from ape.utils import get_all_files_in_directory, get_relative_path
+from ape.utils import cached_property, get_all_files_in_directory, get_relative_path
 
 
-class BaseProject(ProjectAPI):
-    created_temporary_config_file: bool = False
+class ProjectSources:
```

Comment:
if not a part of public API, should it be `_ProjectSources`?

Reply:
[f9e07b9](https://github.com/ApeWorX/ape/pull/1147/commits/f9e07b95089324fd28e7ce3cbf86d48a20e826b4)